src/ui/state_manager.cc

```diff
@@ -0,0 +1,48 @@
+#include "state_manager.hh"
+#include <QTimer>
+#include <QSettings>
+
+StateManager::StateManager( QObject * parent, Config::Class & cfg ):
+  QObject( parent ),
+  _cfg( cfg ),
+  _settings( Config::getStateFileName(), QSettings::IniFormat, parent )
+{
+  _state    = _cfg.mainWindowState;
+  _geometry = _cfg.mainWindowGeometry;
+  _dirty    = false;
+}
+
+
+void StateManager::setState( const QByteArray & state, const QByteArray & geometry )
+{
+  if ( state == _state && _geometry == geometry ) {
+    return;
+  }
+  _state    = state;
+  _geometry = geometry;
+  //in the save state process already.
+  if ( _dirty )
+    return;
+  _dirty = true;
+
+  //delay execution
+  QTimer::singleShot( 3000, this, [ this ]() {
+    saveConfigData( _state, _geometry );
+
+    _dirty = false;
+  } );
+}
+
+
+void StateManager::saveConfigData( QByteArray state, QByteArray geometry )
+{
+  try {
+    _settings.beginGroup( "mainwindow" );
+    _settings.setValue( "state", state );
+    _settings.setValue( "geometry", geometry );
+    _settings.endGroup();
+  }
+  catch ( std::exception & e ) {
```

Comment:
Qt itself never uses exception for error handling. This try/catch doesn't do anything.

I think here should combine QSettings::sync & QSettings::status to write data to disk & check error.

https://doc.qt.io/qt-6/qsettings.html#status

Reply:
![image](https://github.com/user-attachments/assets/8fe6bb32-9099-4bbf-822c-027c36815007)